backend/btrixcloud/storages.py

```diff
@@ -521,6 +531,73 @@ async def get_presigned_url(
 
         return presigned_url, now + self.signed_duration_delta
 
+    async def get_presigned_urls_bulk(
+        self, org: Organization, s3storage: S3Storage, filenames: list[str]
+    ) -> tuple[list[str], datetime]:
+        """generate pre-signed url for crawl file"""
+
+        urls = []
+
+        futures = []
+        num_batch = self.presign_batch_size
+
+        now = dt_now()
+
+        async with self.get_s3_client(
+            s3storage,
+            for_presign=True,
+        ) as (client, bucket, key):
+
+            if (
+                s3storage.access_endpoint_url
+                and s3storage.access_endpoint_url != s3storage.endpoint_url
+            ):
+                parts = urlsplit(s3storage.endpoint_url)
+                host_endpoint_url = f"{parts.scheme}://{bucket}.{parts.netloc}/{key}"
```

Comment:
Might require a change after merging s3v4 change?

Reply:
yes, definitely, perhaps should merge that first then..

Reply:
Refactored to get_host_endpoint_url() to compute this